api/src/main/java/org/openmrs/module/pihapps/PihAppsServiceImpl.java

```diff
@@ -68,4 +100,123 @@ public void updateVisitAndLoginLocations(List<Location> visitLocations, List<Loc
 			}
 		}
 	}
+
+	@Override
+	@Transactional(readOnly = true)
+	@Authorized(PrivilegeConstants.GET_ORDER_TYPES)
+	public List<OrderType> getLabOrderTypes(boolean includeRetired) {
+		List<OrderType> orderTypes = new ArrayList<>();
+		for (OrderType orderType : orderService.getOrderTypes(includeRetired)) {
+			try {
+				Class<?> orderClass = Context.loadClass(orderType.getJavaClassName());
+				if (TestOrder.class.isAssignableFrom(orderClass)) {
```

Comment:
I don't recall exactly what we do in the OWA, but we only want to show Test Orders in this view, not Radiology Test Orders or Pathology Test Orders (both of which I believe would be returned by this call?)

Reply:
Yes, the OWA limits the orders based on order type (defined by the global property `labworkflowowa.testOrderType`), so we probably should do that.  I'm not 100% sure it's always going to be correct, but I believe it is always the way it is done now, so should be fine.

Reply:
OK.  I was trying to be cute with it and see if we could get away without the GP, but I have updated the code to use the same property as used by the lab order component (so we aren't using the same GP as before, but it should be the same value).